rfc/rfc006-distributed-registry.md

```diff
@@ -1,378 +1,582 @@
+    / "d" / "e" / "f" / "g" / "h" / "i" / "j" / "k" / "m" / "n" / "o" / "p"
+    / "q" / "r" / "s" / "t" / "u" / "v" / "w" / "x" / "y" / "z"
+    
+```
 
-`registry/vendor;o=update`
+Where the `idstring` is derived from the public key:
 
-Other values than `update` or `create` for the `o` parameter SHALL NOT be used.
+`idstring = BASE-58(SHA-256(raw-public-key-bytes))`
 
-## 4.1. Creating and updating
-When creating entities its required fields MUST be specified as JSON document. The `o` parameter MUST be specified as `create`.
+For example, consider the following Ed25519 key (as JWK):
 
-To update an object's fields [JSON Patch (RFC6902)](https://tools.ietf.org/html/rfc6902) is used. The object to update
-MUST be specified by specifying the **tid** (*timeline ID*, see [RFC004](rfc004-distributed-document-format.md))
-The `o` parameter MUST be specified as `update`. An update document MUST contain the following fields:
-* **patch** MUST contain an array with patch operations according to [RFC6902](https://tools.ietf.org/html/rfc6902).
+```json
+{
+  "kty" : "OKP",
+  "crv" : "Ed25519",
+  "x"   : "11qYAYKxCrfVS_7TyWQHOg7hcvPapiMlrwIaaPcHURo",
+  "use" : "sig",
+  "kid" : "FdFYFzERwC2uCBB46pZQi4GG85LujR8obt-KWRBICVQ"
+}
+```
 
-If one of the operations fails the document MUST be rejected.
+For this key type the `x` parameter is used to derive `idstring`:
+`idstring = BASE-59(SHA-256(BASE64URL-DECODE(key.x)))`
 
-Example for document type `registry/organization;o=update`:
+Outputs:
 ```json
 {
-    "patch": [
-       { "op": "replace", "path": "/name", "value": "NoPlaceLikeHome Thuiszorg" }
-    ]
+  "id": "did:nuts:e3cacd5c2d931295a64f6c3bb3f6ea58c3a9b253b990e32c5abce43c2f94c564"
 }
 ```
 
-When using a destructive operation (especially when using `remove` or `replace` on arrays) vendors COULD use the
-`test` operation to make sure the intended values are removed.
+Nuts DID Documents are wrapped in a JWS (JSON Web Signature) to ensure cryptographic authenticity and integrity through
+([RFC004](rfc004-distributed-document-format.md)). Please refer to that RFC on how to create the JWS.
 
-## 4.2. Signing
-All documents (registration or update) MUST be signed by the vendor who owns the object (be it a vendor, organization or endpoint). The only
-exception is when registering a new vendor; in that case it MUST be another, already registered vendor. If a new CA certificate
-is registered for a vendor, signing certificate issued by that CA certificate can also be used for updating previously
-published documents from that point on the DAG (see [RFC004](rfc004-distributed-document-format.md)).
+### 3.1 Namespace Specific Identifier (NSI)
 
-In any case the signing certificate MUST conform to [RFC008](rfc008-certificate-structure.md).
+Identifiers are derived from public keys that are valid at the moment of creating the DID document. 
+It MUST be the public key that corresponds to the private key that was used to sign the JWS.
```

Comment:
"It MUST be" must be exactly equal, or just specific fields? How should matching be performed, no canonicalization?

Maybe match on thumbprint?